test/integration/utility.h

```diff
@@ -22,7 +23,10 @@ class BufferingStreamDecoder : public Http::StreamDecoder, public Http::StreamCa
   BufferingStreamDecoder(std::function<void()> on_complete_cb) : on_complete_cb_(on_complete_cb) {}
 
   bool complete() { return complete_; }
-  const Http::HeaderMap& headers() { return *headers_; }
+  const Http::HeaderMap& headers() {
+    ASSERT(headers_ != nullptr);
```

Comment:
IMO a segfaulting test is not the end of the world as it failed. :) I would probably just do nothing, but if you want to do something more clear I would use a gtest assert rather than the prod assert. So ASSET_NE or something.